doc/jemalloc.xml.in

```diff
@@ -1553,6 +1553,24 @@ malloc_conf = "xmalloc:true";]]></programlisting>
         </para></listitem>
       </varlistentry>
 
+      <varlistentry id="opt.prof_leak_error">
+        <term>
+          <mallctl>opt.prof_leak_error</mallctl>
+          (<type>bool</type>)
+          <literal>r-</literal>
+          [<option>--enable-prof</option>]
+        </term>
+        <listitem><para>Leak exit error enabled/disabled.  If enabled, use an
+        <citerefentry><refentrytitle>atexit</refentrytitle>
+        <manvolnum>3</manvolnum></citerefentry> function to report memory leaks
+        detected by allocation sampling, and exit 1 when memory leak is found.  See the
+        <link linkend="opt.prof"><mallctl>opt.prof</mallctl></link> option for
+        information on analyzing heap profile output.  Works only when combined
+        with <link linkend="opt.prof_leak"><mallctl>opt.prof_leak</mallctl>
+        </link>, otherwise does nothing.  This option is disabled by default.
```

Comment:
> otherwise does nothing

That worries me a little. We have a lot of options, the more we add, the more people are unlikely to read the descriptions carefully. If we leave it like that, some people can miss this detail and still use the option, thinking it would detect leaks.

I would add validation in `jemalloc.c:malloc_conf_init_helper` and abort if `prof_leak_error` is enabled and `prof_leak` is disabled. @interwq what do you think?

Reply:
> abort if prof_leak_error is enabled and prof_leak is disabled

Agreed, i.e. when `abort_conf == true`, don't allow setting prof_leak_error w/o prof_leak.